applications/asset_tracker/src/main.c

```diff
@@ -1850,5 +1871,12 @@ void main(void)
 	k_sem_take(&cloud_ready_to_connect, K_FOREVER);
 #endif
 
+	date_time_update_async();
+
+	/** Sleep in order to ensure that the date time library has obtained
+	 *  time before connecting/publishing to cloud.
+	 */
+	k_sleep(K_SECONDS(5));
```

Comment:
Do we have a better way to solve this? A way to confirm time was updated?

Reply:
Agree on that

Reply:
I have added a callback handler in the module. The application now depends on a semaphore released upon a callback response from the library.